### PR TITLE
Symfony 2.8 Error was fixed Catchable Fatal Error: Argument 1 passed …

### PR DESCRIPTION
**Pull request by AndreySyagrovskiy**
…to Sonata\AdminBundle\Form\DataTransformer\ModelsToArrayTransformer::__construct() must be an instance of Sonata\AdminBundle\Form\ChoiceList\ModelChoiceList, instance of Symfony\Component\Form\ChoiceList\LazyChoiceList given, called in /home/andrey/sites/renmark/app/cache/dev/classes.php on line 14324 and defined

### CODE REVIEW

**Comment by OskarStark** [comment]
can you please add a test for the ``LazyChoiceList`` ?

File: Form/DataTransformer/ModelsToArrayTransformer.php [hunk 1]
```diff
@@ -42,11 +43,11 @@ class ModelsToArrayTransformer implements DataTransformerInterface
...
      *
-     * @param ModelChoiceList       $choiceList
+     * @param ModelChoiceList|LazyChoiceList $choiceList
      * @param ModelManagerInterface $modelManager
      * @param                       $class
      */
-    public function __construct(ModelChoiceList $choiceList, ModelManagerInterface $modelManager, $class)
+    public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
     {
         $this->choiceList   = $choiceList;
```
Comment (Koc) [inline Form/DataTransformer/ModelsToArrayTransformer.php]: Please add checks using `instanceof` that `$choiceList` is `ModelChoiceList` or `LazyChoiceList `but not an arbitrary object

File: composer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "sonata-project/admin-bundle",
+    "name": "sonata-project-saygrfix/admin-bundle",
```
Comment (Koc) [inline composer.json]: Please revert changes in this file, include permissions.

**Comment by Koc** [comment]
related to #3480

**Comment by Storke** [comment]
#3480 corrects the way "children" property is accessed on FormBuilder, this is not related to this PR.

**Comment by rande** [comment]
fixed by https://github.com/sonata-project/SonataAdminBundle/pull/3504

**Pull request closed without merging**